### ISSUE TITLE
Support eslint `dot-notation` rule  for enum type

### ISSUE THREAD

**Issue by iFwu** [typescript-eslint/typescript-eslint#474]
[`dot-notation`](https://eslint.org/docs/rules/dot-notation) rule is in the eslint core rule, however, it doesn't work for linting enum type in ts files.

**Comment by bradzacher** [comment]
Hi @iFwu - could you please give us some more information?
We have an issue template for a reason - to get all the information up front to expedite the process.

Please fill out the below template:

    

    **Repro**

    

    ```JSON
    {
      "rules": {
        "typescript/<rule>": "<setting>"
      }
    }
    ```

    ```TS
    // your repro code case
    ```

    **Expected Result**

    **Actual Result**

    **Additional Info**

    **Versions**

    | package                            | version |
    | ---------------------------------- | ------- |
    | `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
    | `@typescript-eslint/parser`        | `X.Y.Z` |
    | `TypeScript`                       | `X.Y.Z` |
    | `ESLint`                           | `X.Y.Z` |
    | `node`                             | `X.Y.Z` |
    | `npm`                              | `X.Y.Z` |
